spark-jobs/src/main/scala/za/co/absa/enceladus/standardization/ControlInfoValidation.scala

```diff
@@ -25,29 +28,46 @@ import scala.util.{Failure, Success, Try}
 object ControlInfoValidation {
   private val rawFieldName = "raw"
   private val sourceFieldName = "source"
-
-  case class FieldCounts(rawCount: Long, sourceCount: Long)
+  private val configEntry = "control.info.validation"
 
   /**
    * Adds metadata about the number of records in raw and source data by checking Atum's checkpoints first.
    */
-  def addRawAndSourceRecordCountsToMetadata(): Unit = {
-    val fields = getFieldCounts(Atum.getControlMeasure.checkpoints)
-    Atum.setAdditionalInfo(s"${rawFieldName}_record_count" -> fields.rawCount.toString)
-    Atum.setAdditionalInfo(s"${sourceFieldName}_record_count" -> fields.sourceCount.toString)
-  }
-
-  def getFieldCounts(checkpoints: Seq[Checkpoint]): FieldCounts = {
+  def addRawAndSourceRecordCountsToMetadata(conf: Config, log: Logger): Unit = {
+    val validationType = conf.getString(configEntry)
+    val checkpoints = Atum.getControlMeasure.checkpoints
     val checkpointRawRecordCount = getCountFromGivenCheckpoint(rawFieldName, checkpoints)
     val checkpointSourceRecordCount = getCountFromGivenCheckpoint(sourceFieldName, checkpoints)
-    val errorMessage = "Checkpoint validation failed:"
+
+    try {
+        validateFields(checkpointRawRecordCount, checkpointSourceRecordCount)
+    } catch {
+      case ex: ValidationException => validationType match {
+        case "strict" => throw ex
+        case "warning" => log.warn(ex.msg)
+        case _ =>
+      }
+    } finally {
+      checkpointRawRecordCount
+        .foreach(e => Atum.setAdditionalInfo(s"${rawFieldName}_record_count" -> e.toString))
+
+      checkpointSourceRecordCount
+        .foreach(e => Atum.setAdditionalInfo(s"${sourceFieldName}_record_count" -> e.toString))
+    }
+  }
+
+  def validateFields(checkpointRawRecordCount: Try[Long], checkpointSourceRecordCount: Try[Long]) {
```

Comment:
Would make it a function returning `Try[Unit]`. Also would make it `private` I don't think this needs such a granular level level of unit-testing.
(I believe it's consistent and better to read if one sticks within one class/flo to either use `class Try` xor `try...catch` block.)